### PR TITLE
JAVA-847: CodecRegistry is not propagated to nested UDTs.

### PR DESCRIPTION
**Pull request by adutra**

### CODE REVIEW

**Comment by tolbertam** [comment]
Looks good to me :+1:

File: driver-core/src/test/java/com/datastax/driver/core/TypeCodecNestedUDTAndTupleIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            "INSERT INTO t1 (pk, c1) VALUES (1, {f1:{f2:'foo'}})",
+            "INSERT INTO t1 (pk, c2) VALUES (2, (('foo')))"
+        );
+    }
+
+    @Test(groups = "short")
+    public void should_set_registry_on_nested_udts() {
+        ResultSet rows = session.execute("SELECT c1 FROM t1 WHERE pk = 1");
+        Row row = rows.one();
+        // here the CodecRegistry will create a codec on-the-fly using the UserType received from the resulset metadata
```
Comment (tolbertam) [inline driver-core/src/test/java/com/datastax/driver/core/TypeCodecNestedUDTAndTupleIntegrationTest.java]: small nitpick here and at line 68, resulset -> resultset

File: driver-core/src/test/java/com/datastax/driver/core/TypeCodecNestedUDTAndTupleIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+ * the registry encounters the UDT or tuple, it's coming from a ResultSet,
+ * not from the Cluster's metadata.
+ *
+ * @jira_ticket JAVA-847
+ */
+@CassandraVersion(major = 2.1)
+public class TypeCodecNestedUDTAndTupleIntegrationTest extends CCMBridge.PerClassSingleNodeCluster {
+
+    @Override
+    protected Collection<String> getTableDefinitions() {
```
Comment (JeremiahDJordan) [inline driver-core/src/test/java/com/datastax/driver/core/TypeCodecNestedUDTAndTupleIntegrationTest.java:40]: Probably good to have a test for a tuple inside a UDT and a UDT inside a tuple, also 3 layers deep of something.

Comment (adutra) [inline driver-core/src/test/java/com/datastax/driver/core/TypeCodecNestedUDTAndTupleIntegrationTest.java:40]: Good idea, I added another test with an even deeper structure.

**Pull request merged**